keyboards/handwired/replicazeron/stm32f103/halconf.h

```diff
@@ -0,0 +1,27 @@
+/* Copyright 2023 9R
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+//enable i2c
+#define HAL_USE_I2C TRUE
+
+#define HAL_USE_PWM TRUE
+
+//enable adc
+#define HAL_USE_ADC TRUE
```

Comment:
With the `(peripheral)_DRIVER_REQUIRED` stuff, it adds the "hal use" defines automatically.  And the oled requires the i2c driver.   So you can remove them, if you don't explicitly need them.  

```suggestion
#define HAL_USE_PWM TRUE
```

Also, I don't see pwm being used here at all?  Did you mean to include it for the ws2812 driver?  If so, it doesn't look like you've enabled that. 

Otherwise, this whole file could be removed.  (if you need the pwm driver then you wouldn't want to remove this)

Reply:
I'm not sure why I put HA_USE_PWM in there anymore. Might be a relic from when I experimented with monochrome LED lighting.
Anyway, I ran some tests applying your changes and removing halconf.h completely and there are no regressions. So you are right: That file is obsolete.